### PR TITLE
plexamp: add darwin pkgs

### PR DESCRIPTION
**Pull request by klchen0112**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/plexamp/linux.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{ lib, fetchurl, appimageTools, pkgs, pname, version, meta, ... }:
+
+let
+  inherit pname version meta;
+  name = "${pname}-${version}";
+  src = fetchurl {
+    url = "https://plexamp.plex.tv/plexamp.plex.tv/desktop/Plexamp-${version}.AppImage";
+    name = "${pname}-${version}.AppImage";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/linux.nix:8]:
```suggestion
```
duplicated with the filename in the url

File: pkgs/applications/audio/plexamp/linux.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    sha512 = "7NQmH42yzItReQ5WPdcbNPr/xed74H4UyDNlX5PGmoJRBpV1RdeuW1KRweIWfYNhw0KeqULVTjr/aCggoWp4WA==";
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/linux.nix:9]: please use sha256 sri encoded hashes

File: pkgs/applications/audio/plexamp/linux.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  };
+
+  appimageContents = appimageTools.extractType2 {
+    inherit name pname version src;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/linux.nix:13]:
```suggestion
    inherit pname version src;
```
providing name and pname+version is unecessary

File: pkgs/applications/audio/plexamp/linux.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  };
+in
+appimageTools.wrapType2 {
+  inherit name pname version src meta;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/linux.nix:17]:
```suggestion
  inherit pname version src meta;
```

File: pkgs/applications/audio/plexamp/linux.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  multiArch = false; # no 32bit needed
+  extraPkgs = pkgs: appimageTools.defaultFhsEnvArgs.multiPkgs pkgs ++ [ pkgs.bash ];
+
+  extraInstallCommands = ''
+    ln -s $out/bin/${pname}-${version} $out/bin/${pname}
+    install -m 444 -D ${appimageContents}/plexamp.desktop $out/share/applications/plexamp.desktop
+    install -m 444 -D ${appimageContents}/plexamp.png \
+      $out/share/icons/hicolor/512x512/apps/plexamp.png
+    substituteInPlace $out/share/applications/${pname}.desktop \
+      --replace 'Exec=AppRun' 'Exec=${pname}'
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/linux.nix:28]: Please do not substitute pname everywhere, it is unnecessary and increases complexity

File: pkgs/applications/audio/plexamp/linux.nix [hunk 6]
```diff
@@ -0,0 +1,32 @@
+{ lib, fetchurl, appimageTools, pkgs, pname, version, meta, ... }:
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/linux.nix:1]:
```suggestion
{ lib, fetchurl, appimageTools, pkgs, pname, version, meta }:
```

File: pkgs/applications/audio/plexamp/default.nix [hunk 7]
```diff
@@ -1,41 +1,22 @@
-{ lib, fetchurl, appimageTools, pkgs }:
+{ lib
+, stdenv
+, callPackage
+, ...
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/default.nix:4]:
```suggestion
```

File: pkgs/applications/audio/plexamp/default.nix [hunk 8]
```diff
@@ -1,41 +1,22 @@
...
     description = "A beautiful Plex music player for audiophiles, curators, and hipsters";
     homepage = "https://plexamp.com/";
     changelog = "https://forums.plex.tv/t/plexamp-release-notes/221280/52";
     license = licenses.unfree;
     maintainers = with maintainers; [ killercup synthetica ];
-    platforms = [ "x86_64-linux" ];
+    platforms = [ "x86_64-linux" "aarch64-darwin" "x86_64-darwin" ];
   };
-}
+  extraArgs = removeAttrs args [ "callPackage" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/default.nix:18]: This is unnecessary

File: pkgs/applications/audio/plexamp/darwin.nix [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+          {
+            url = "https://plexamp.plex.tv/plexamp.plex.tv/desktop/Plexamp-${version}.dmg";
+            sha256 = "sha256-gXJySo/8LwDkyR0mJNou5BD+7oxx6Crmcatf/uyMTjQ=";
+          }
+      );
+
+  dontPatch = true;
+  dontConfigure = true;
+  dontBuild = true;
+  dontFixup = true;
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/darwin.nix:32]:
```suggestion
  dontBuild = true;
```

disabling patches here just complicates things when users want to apply patches locally and it doesn't break things. Probably the same with configure.

Why not fixup?

File: pkgs/applications/audio/plexamp/darwin.nix [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+          sha256 = "sha256-VmADu5vC2SZs930mgSw0uF69xSsPfTy/1feBYTHM55g=";
+        })
+    else
+      (
+        fetchurl
+          {
+            url = "https://plexamp.plex.tv/plexamp.plex.tv/desktop/Plexamp-${version}.dmg";
+            sha256 = "sha256-gXJySo/8LwDkyR0mJNou5BD+7oxx6Crmcatf/uyMTjQ=";
+          }
+      );
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/darwin.nix:27]:
```suggestion
  src = if stdenv.isAarch64 then 
    fetchurl {
        url = "https://plexamp.plex.tv/plexamp.plex.tv/desktop/Plexamp-${version}-arm64.dmg";
        sha256 = "sha256-VmADu5vC2SZs930mgSw0uF69xSsPfTy/1feBYTHM55g=";
      }
    else
      fetchurl {
        url = "https://plexamp.plex.tv/plexamp.plex.tv/desktop/Plexamp-${version}.dmg";
        sha256 = "sha256-gXJySo/8LwDkyR0mJNou5BD+7oxx6Crmcatf/uyMTjQ=";
      };
```

File: pkgs/applications/audio/plexamp/darwin.nix [hunk 11]
```diff
@@ -0,0 +1,42 @@
+{ lib
+, stdenv
+, fetchurl
+, undmg
+, pname
+, version
+, meta
+, ...
```
Comment (SuperSandro2000) [inline pkgs/applications/audio/plexamp/darwin.nix:8]:
```suggestion
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/plexamp/linux.nix (hunk 2 above)
Comment (klchen0112) [reply pkgs/applications/audio/plexamp/linux.nix:9]: update-plexamp.sh is used to update the sha512 in default.nix. So  if i change to sha 256 , update-plexamp.sh could remove

File: pkgs/applications/audio/plexamp/darwin.nix (hunk 9 above)
Comment (klchen0112) [reply pkgs/applications/audio/plexamp/darwin.nix]: sorry, t I'm not very familiar with building a nixpkg

**Pull request closed without merging**